app/lib/service/download_counts/sync_download_counts.dart

```diff
@@ -56,120 +56,157 @@ Map<String, int> _extractDayCounts(Map<String, dynamic> json) {
   return dayCounts;
 }
 
-Future<bool> processDownloadCounts(
-    String downloadCountsFileName, DateTime date) async {
-  final downloadCountsBucket =
+Future<Set<String>> processDownloadCounts(DateTime date) async {
+  final downloadCountsFileName = [
+    'daily_download_counts',
+    formatDateForFileName(date),
+    'data-000000000000.jsonl',
+  ].join('/');
+
+  final fileNamePrefix =
+      ['daily_download_counts', formatDateForFileName(date), 'data-'].join('/');
+  final bucket =
       storageService.bucket(activeConfiguration.downloadCountsBucketName!);
-  Uint8List bytes;
-  try {
-    bytes = await downloadCountsBucket.readAsBytes(downloadCountsFileName);
-  } on Exception catch (e) {
-    _logger.info('Failed to read "$downloadCountsFileName"./n'
-        '$e');
-    return false;
-  }
-  if (bytes.isEmpty) {
-    _logger.severe('$downloadCountsFileName is empty.');
-    return false;
-  }
 
-  bool hasFailedLines = false;
-  // Before '2024-05-03' the query generating the download count data had a bug
-  // in the reg exp causing incorrect versions to be stored.
-  final regExpQueryFixDate = DateTime.parse('2024-05-03');
-  final processedPackages = <String>{};
-  var lines = <String>[];
-
-  try {
-    lines = utf8.decode(bytes).split('\n');
-  } on FormatException catch (e) {
-    _logger.severe('Failed to utf8 decode bytes of $downloadCountsFileName/n'
-        '$e');
-    return false;
+  final failedFiles = <String>{};
+
+  if (await bucket.list(prefix: fileNamePrefix).isEmpty) {
+    _logger.info('Failed to read any files with prefix "$fileNamePrefix"./n');
+    failedFiles.add(fileNamePrefix);
   }
 
+  final processedPackages = <String>{};
   final pool = Pool(10);
-  await Future.wait(lines.map((line) async {
-    return await pool.withResource(() async {
-      if (line.isBlank) {
-        return;
-      }
-      final String package;
-      final Map<String, int> dayCounts;
-      try {
-        final data = json.decode(line);
-        if (data is! Map<String, dynamic>) {
-          throw FormatException('Download counts data is not valid json');
-        }
+  await for (final f in bucket.list(prefix: fileNamePrefix)) {
```

Comment:
Try to only invoke bucket.list once

Maybe do `bucket.list(...).toList();`

Reply:
done